vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_embedding_info.h

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_PIPELINE_PIPELINE_EMBEDDING_INFO_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_PIPELINE_PIPELINE_EMBEDDING_INFO_H_
+
+#include <map>
+#include <string>
+
+#include "bat/ads/internal/ml/data/vector_data.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+struct PipelineEmbeddingInfo final {
+  PipelineEmbeddingInfo();
+  PipelineEmbeddingInfo(PipelineEmbeddingInfo&& info) noexcept;
+  PipelineEmbeddingInfo& operator=(PipelineEmbeddingInfo&& info) noexcept;
+  ~PipelineEmbeddingInfo();
+
+  PipelineEmbeddingInfo(const int version,
+                        const std::string& timestamp,
+                        const std::string& locale,
+                        const int embeddings_dim,
+                        const std::map<std::string, VectorData>& embeddings);
+
+  int version;
+  std::string timestamp;
+  std::string locale;
+  int embeddings_dim;
+  std::map<std::string, VectorData> embeddings;
```

Comment:
could we call this `vocabulary_embeddings`? (bringing over from previous PR)

Reply:
this is a general pipeline object, which can be used beyond vocab embeddings so maybe keep as is?

Reply:
I vote to keep it as is. If we later require embedding something that is non-text (e.g. an image), I'd opt to have a different pipeline object (e.g. pipeline_image_embedding). So calling the attribute vocabulary_embeddings or text_embeddings seems redundant to me.